src/kagglehub/gcs_upload.py

```diff
@@ -0,0 +1,156 @@
+import logging
+import os
+from datetime import datetime
+
+import requests
+
+from kagglehub.clients import KaggleApiV1Client
+from kagglehub.exceptions import BackendError, postprocess_response
+
+logger = logging.getLogger(__name__)
+
+MAX_NUM_FILES = 50
+
+
+def parse_datetime_string(string: str):
+    time_formats = ["%Y-%m-%dT%H:%M:%S", "%Y-%m-%dT%H:%M:%SZ", "%Y-%m-%dT%H:%M:%S.%f", "%Y-%m-%dT%H:%M:%S.%fZ"]
+    for t in time_formats:
+        try:
+            result = datetime.strptime(string[:26], t).replace(microsecond=0)  # noqa: DTZ007
+            return result
+        except:  # noqa: E722, S110
+            pass
+    return string
+
+
+class File(object):  # noqa: UP004
```

Comment:
Where do you use this class?

I am ctrl-F and can't find any usage.

Reply:
its used in the upload_file function:
```
token = _upload_blob(full_path, model_type)
    if not quiet:
        logger.info("Upload successful: " + file_name + " (" + File.get_size(content_length) + ")")
    return token
```

Reply:
Ah. Makes sense. I was CTRL-f'ing for `File(`.

Reply:
Are you instantiate a File object anywhere? If not, then, the `__init__` method isn't used.

Looks like only the static `File.get_size()` is used. Consider just having a `def humanize_bytes(...)` or something like that.